### PR TITLE
`PlayerMeleeWeaponAnimation` enum

### PR DESCRIPTION
**Pull request by Dragorn421**
`Player.meleeWeaponAnimation` can take values corresponding to the current melee weapon animation, like when swinging swords, hammer, stick

I moved the enum from Dark Link (`z_en_torch2`) so shout out to @petrie911 for making the enum in the first place

Suggestions on the prefix are welcome. Currently it is `PLAYER_MWA_`, MWA standing for Melee Weapon Animation. We already have this with `PLAYER_AP_` for Action Parameter and I think it's a good trend to follow

------------

I checked the enum names in game with the following code in `Player_Draw`

<details>

```c
{
char* names[] = {
    "PLAYER_MWA_FORWARD_SLASH_1H", "PLAYER_MWA_FORWARD_SLASH_2H", "PLAYER_MWA_FORWARD_COMBO_1H",
    "PLAYER_MWA_FORWARD_COMBO_2H", "PLAYER_MWA_RIGHT_SLASH_1H",   "PLAYER_MWA_RIGHT_SLASH_2H",
    "PLAYER_MWA_RIGHT_COMBO_1H",   "PLAYER_MWA_RIGHT_COMBO_2H",   "PLAYER_MWA_LEFT_SLASH_1H",
    "PLAYER_MWA_LEFT_SLASH_2H",    "PLAYER_MWA_LEFT_COMBO_1H",    "PLAYER_MWA_LEFT_COMBO_2H",
    "PLAYER_MWA_STAB_1H",          "PLAYER_MWA_STAB_2H",          "PLAYER_MWA_STAB_COMBO_1H",
    "PLAYER_MWA_STAB_COMBO_2H",    "PLAYER_MWA_FLIPSLASH_START",  "PLAYER_MWA_JUMPSLASH_START",
    "PLAYER_MWA_FLIPSLASH_FINISH", "PLAYER_MWA_JUMPSLASH_FINISH", "PLAYER_MWA_BACKSLASH_RIGHT",
    "PLAYER_MWA_BACKSLASH_LEFT",   "PLAYER_MWA_HAMMER_FORWARD",   "PLAYER_MWA_HAMMER_SIDE",
    "PLAYER_MWA_SPIN_ATTACK_1H",   "PLAYER_MWA_SPIN_ATTACK_2H",   "PLAYER_MWA_BIG_SPIN_1H",
    "PLAYER_MWA_BIG_SPIN_2H",
};
GfxPrint printer;
Gfx* gfx;
Player* player = GET_PLAYER(globalCtx);

OPEN_DISPS(globalCtx->state.gfxCtx, __FILE__, __LINE__);

gfx = POLY_OPA_DISP + 1;
gSPDisplayList(OVERLAY_DISP++, gfx);

GfxPrint_Init(&printer);
GfxPrint_Open(&printer, gfx);

GfxPrint_SetColor(&printer, 255, 0, 255, 255);
GfxPrint_SetPos(&printer, 10, 10);
GfxPrint_Printf(&printer, "%d %s", player->meleeWeaponAnimation, names[player->meleeWeaponAnimation]);

gfx = GfxPrint_Close(&printer);
GfxPrint_Destroy(&printer);

gSPEndDisplayList(gfx++);
gSPBranchList(POLY_OPA_DISP, gfx);
POLY_OPA_DISP = gfx;

CLOSE_DISPS(globalCtx->state.gfxCtx, __FILE__, __LINE__);
}
```

</details>

~~I never tinkered with the sword controls/abilities before so maybe I'm just a noob but I couldn't trigger:~~
~~`PLAYER_MWA_LEFT_*`, `PLAYER_MWA_STAB_2H`, `PLAYER_MWA_STAB_COMBO_*`~~ (those rely on ztargetting), ~~`PLAYER_MWA_FLIPSLASH_*`, `PLAYER_MWA_BACKSLASH_*`~~ (those are unused)

The other names seemed accurate, except maybe `PLAYER_MWA_RIGHT_` because it makes Link slash right-to-left and then left-to-right on combo, so I'm not sure what "right" means

~~Also `PLAYER_MWA_HAMMER_SIDE` is kinda hard to trigger I found, is there a consistent way to get that?~~ (yes, with ztargetting it's far easier)

### CODE REVIEW

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 1]
```diff
@@ -3287,25 +3327,25 @@ s32 func_80837818(Player* this) {
...
         if (func_808375D8(this)) {
-            sp18 = 24;
+            sp18 = PLAYER_MWA_SPIN_ATTACK_1H;
         } else {
             if (sp1C < 0) {
                 if (func_80833BCC(this)) {
-                    sp18 = 0;
+                    sp18 = PLAYER_MWA_FORWARD_SLASH_1H;
                 } else {
                     sp18 = 4;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_player_actor/z_player.c:3336]: Missed this one?

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 2]
```diff
@@ -1029,34 +1029,62 @@ static ExplosiveInfo sExplosiveInfos[] = {
     { ITEM_BOMBCHU, ACTOR_EN_BOM_CHU },
 };
 
-static struct_80854190 D_80854190[] = {
+static struct_80854190 D_80854190[PLAYER_MWA_MAX] = {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_player_actor/z_player.c:1032]: This and a number of other structs can be named too.

**Comment by EllipticEllipsis** [comment]
The flip jumpslash and the backslash are well-known as being unused. I did some research on this with a watch in gz when compiling the information on [this spreadsheet](https://docs.google.com/spreadsheets/d/12RpxLB-AhiilGi6aG7MNKia2lhkAKivjly9V8GL-utg/edit?usp=sharing), and I think the descriptions there probably explain how to trigger these: to get the rightwalk ones, you have to target and walk right. All of the third ones work the same way: you time the slashes correctly and the third one uses the animations 2 entries down (I assume this is in the code somewhere, although it doesn't stick out immediately.)

Also, the 1H/2H thing, if we keep it, must absolutely be explained somewhere: I've looked at this system before and I didn't immediately understand what it meant, so I don't think other people are going to have much chance. Given that the third slash

**Comment by fig02** [comment]
I haved't reviewed the PR yet but I think we really need to call it something other than an animation. slash type or something

**Review by fig02 — APPROVED** [review]
I dont think i like melee weapon action that much, but we can get this in and decide on a name later

**Review by Roman971 — APPROVED** [review]

**Comment by fig02** [comment]
good to merge after conflict is resolved

**Pull request merged**